src/crystal/system/win32/delay_load.cr

```diff
@@ -15,6 +8,21 @@ private macro p_from_rva(rva)
   pointerof(LibC.image_base).as(UInt8*) + {{ rva }}
 end
 
+private macro print_error(format, *args)
+  {% if args.empty? %}
+    %str = {{ format }}
+    LibC.WriteFile(LibC.GetStdHandle(LibC::STD_ERROR_HANDLE), %str, %str.bytesize, out _, nil)
+  {% else %}
+    %buf = uninitialized LibC::CHAR[1024]
+    %args = uninitialized UInt64[{{ args.size }}]
```

Comment:
According to the documentation of `FormatMessageA` with `FORMAT_MESSAGE_ARGUMENT_ARRAY` the type of the arguments argument should be `DWORD_PTR`.

```suggestion
    %args = uninitialized LibC::DWORD[{{ args.size }}]
```

But I figure then the 64-bit address values won't work anymore...? The docs explicitly state that this flag doesn't work for 64-bit integers and you need to use a `va_list` instead.

Reply:
`DWORD_PTR` is not a pointer to `DWORD`, that would be `PDWORD` instead.

Reply:
Ah thanks, I didn't realize that.
However I'm still confused about the docs saying that 64-bit integer values are not supported when addresses are clearly 64-bit integers... 🤔

Reply:
Shouldn't this be something like SizeT or another universal alias rather than hardcoding 64-bit pointers?

@straight-shoota I think it doesn't want you to point to a 64-bit integer. Not sure if there are any here.

Reply:
Or even how about just turning this into `StaticArray(Void*)` altogether

Reply:
The parameter type is a void pointer. That's the static array with a value type that could be anything. The value is expected to be `DWORD_PTR`, so `UInt64` seems correct. It wouldn't hurt to introduce ´LibC::DWORD_PTR` for clarity, though.